symja_android_library/matheclipse-core/src/main/java/org/matheclipse/core/expression/AbstractFractionSym.java

```diff
@@ -56,13 +59,15 @@
    * @param maxConbvergents maximum number of convergents.
    * @return stream of {@link BigFraction} convergents approximating {@code value}
    */
-  public static Stream<BigFraction> convergents(final double value, final int maxConvergents) {
+  // TODO: replace this by Hipparcus BigFraction.convergents() after next Hipparchus release
+  public static Stream<BigFraction> convergents(double value, int maxConvergents) {
     if (FastMath.abs(value) > Integer.MAX_VALUE) {
       throw new MathIllegalStateException(LocalizedCoreFormats.FRACTION_CONVERSION_OVERFLOW, value,
           value, 1l);
     }
-    return StreamSupport.stream(Spliterators.spliteratorUnknownSize(
-        generatingIterator(value, maxConvergents), Spliterator.DISTINCT), false);
+    return StreamSupport
+        .stream(Spliterators.spliteratorUnknownSize(generatingIterator(value, maxConvergents),
+            Spliterator.DISTINCT | Spliterator.ORDERED), false);
```

Comment:
@axkr do you know why the spliterator does not yet report the ordered characteristics? I'm not sure but I think it should do that because the generated fractions have an order and should not be arbitrarily reordered.